components/WorkDatesDisplay/WorkDatesDisplay.js

```diff
@@ -1,33 +1,47 @@
 import React, { Fragment } from "react";
 import { DatesDisplay, DeleteButton, Dates } from "./WorkDatesDisplay.styled";
+import { format } from "date-fns";
 
-export default function WorkDatesDisplay({ workDates, onDateRemove }) {
+export default function WorkDatesDisplay({ absenceDates, onDateRemove }) {
+  const vacationDates =
+    absenceDates?.filter((absence) => absence.type === "vacation") || [];
+  const daysOff =
+    absenceDates?.filter((absence) => absence.type === "dayOff") || [];
+  console.log(vacationDates);
+  console.log(daysOff);
   return (
     <DatesDisplay>
       <h4>Your Vacation Dates:</h4>
-      {workDates && workDates.length > 0 ? (
-        <ul>
-          {workDates.map((workDate) => (
-            <li key={workDate._id}>
-              {workDate.vacationDates.map((dateRange, index) => (
-                <Fragment key={index}>
-                  <Dates>
-                    From: {new Date(dateRange.startDate).toLocaleDateString()}
-                    To: {new Date(dateRange.endDate).toLocaleDateString()}
-                  </Dates>
-                  <DeleteButton
-                    onClick={() => onDateRemove(index, workDate._id)}
-                  >
-                    Remove
-                  </DeleteButton>
-                </Fragment>
-              ))}
-            </li>
-          ))}
-        </ul>
-      ) : (
-        <p>No Vacations yet!</p>
-      )}
+      <ul>
+        {vacationDates.map((vacation) => (
+          <Fragment key={vacation._id}>
+            {vacation.date.map((date, index) => (
+              <li key={index}>
```

Comment:
The `index` is not enough for the `key` prop. Use a stable and unique identifier instead. You could use the date itself if you don't have an id.